### PR TITLE
Upgrade wp-prettier to 2.6.2

### PR DESCRIPTION
**Pull request by jsnajdr**
I updated the `wp-prettier` fork to be in sync with the latest upstream version, Prettier 2.6.2, and published it as `wp-prettier@2.6.2`.

This PR updates the Gutenberg dependency and reformats the codebase.

### CODE REVIEW

**Review by dmsnell — APPROVED** [review]
Couldn't review any changes on this page because GitHub kept freezing the browser when trying to show the diff, but the changes to the parser code look good 👍

File: packages/scripts/CHANGELOG.md [hunk 1]
```diff
@@ -2,10 +2,14 @@
 
 ## Unreleased
 
-## Breaking Changes
+### Breaking Changes
```
Comment (ntwb) [inline packages/scripts/CHANGELOG.md]: This is somewhat clever that Prettier caught this

File: packages/eslint-plugin/utils/constants.js [hunk 2]
```diff
@@ -36,7 +36,8 @@ const TRANSLATION_FUNCTIONS = new Set( [ '__', '_x', '_n', '_nx' ] );
  *
  * @type {RegExp}
  */
-const REGEXP_SPRINTF_PLACEHOLDER = /%(((\d+)\$)|(\(([$_a-zA-Z][$_a-zA-Z0-9]*)\)))?[ +0#-]*\d*(\.(\d+|\*))?(ll|[lhqL])?([cduxXefgsp%])/g;
+const REGEXP_SPRINTF_PLACEHOLDER =
```
Comment (ntwb) [inline packages/eslint-plugin/utils/constants.js:39]: The regex explainer comments alignment below this line can be fixed in a follow up PR

File: packages/dependency-extraction-webpack-plugin/lib/types.d.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ declare interface DependencyExtractionWebpackPluginOptions {
 	injectPolyfill?: boolean;
 	useDefaults?: boolean;
 	outputFormat?: 'php' | 'json';
-	outputFilename?: string | Function,
+	outputFilename?: string | Function;
```
Comment (ntwb) [inline packages/dependency-extraction-webpack-plugin/lib/types.d.ts]: Nice catch 👌🏼

File: packages/components/src/icon/index.tsx [hunk 4]
```diff
@@ -68,10 +68,10 @@ function Icon< P >( {
 
 	if ( 'function' === typeof icon ) {
 		if ( icon.prototype instanceof Component ) {
-			return createElement( icon, ( {
+			return createElement( icon, {
 				size,
 				...additionalProps,
-			} as unknown ) as P );
+			} as unknown as P );
```
Comment (ntwb) [inline packages/components/src/icon/index.tsx:74]: Parenthesis removed

**Review by ntwb — COMMENTED** [review]
Took a little while,eventually got through all the files, the vast majorjty, 99% are line length changes, shorter, or longer

Used the hide whitespace view to review https://github.com/WordPress/gutenberg/pull/40542/files?w=1

I commented on 4 file changes out of the 568 files, so looks pretty good to me

File: packages/components/src/icon/index.tsx (hunk 4 above)
Comment (jsnajdr) [reply packages/components/src/icon/index.tsx:74]: Yes, that's what the updated Prettier does to all TypeScript `X as Y as Z` expressions. There are no longer any parens.

File: packages/eslint-plugin/utils/constants.js (hunk 2 above)
Comment (jsnajdr) [reply packages/eslint-plugin/utils/constants.js:39]: Fixed in an extra commit, thanks for spotting this :+1:

File: packages/scripts/CHANGELOG.md (hunk 1 above)
Comment (jsnajdr) [reply packages/scripts/CHANGELOG.md]: No, this was me when I was adding a Prettier upgrade notice to the `@wordpress/scripts` changelog 🙂

**Comment by jsnajdr** [comment]
Thanks @ntwb and @dmsnell for going through the changes! @gziolo recommended to wait with merging this PR until the WordPress 6.0 release cycle is done, so that the formatting changes don't complicate cherry-picking individual patches.

File: packages/prettier-config/lib/index.js [hunk 5]
```diff
@@ -13,7 +13,7 @@ const prettierPackage = require( require.resolve( 'prettier/package.json' ) );
 
 const isWPPrettier = prettierPackage.name === 'wp-prettier';
 const customOptions = isWPPrettier
-	? { parenSpacing: true, jsxBracketSameLine: false }
+	? { parenSpacing: true, bracketSameLine: false }
```
Comment (gziolo) [inline packages/prettier-config/lib/index.js]: `bracketSameLine` is the same now in both cases so we can move it to the `config` object.

Comment (jsnajdr) [reply packages/prettier-config/lib/index.js]: Cleaned up :+1:

**Comment by noisysocks** [comment]
Can we add the merge commit to `.git-blame-ignore-revs` once merged? https://github.com/WordPress/gutenberg/blob/trunk/.git-blame-ignore-revs

**Comment by jsnajdr** [comment]
> Can we add the merge commit to `.git-blame-ignore-revs`

Will do! Thanks for mentioning this, I didn't know that `.git-blame-ignore-revs` is a thing.

**Comment by gziolo** [comment]
This PR requires another Prettier run to format files changed in the `trunk` in the meantime. Otherwise, it should be good to land. WordPress 6.0 is out so it's no longer a concern.

**Review by gziolo — APPROVED** [review]
Let's move forward with this Prettier fork's upgrade. It will require another rebase and Prettier formatting run before landing.

> Can we add the merge commit to .git-blame-ignore-revs once merged?
> https://github.com/WordPress/gutenberg/blob/trunk/.git-blame-ignore-revs

Let's remember about the follow-up suggested by @noisysocks.

**Comment by gziolo** [comment]
> > Can we add the merge commit to .git-blame-ignore-revs once merged?
> > https://github.com/WordPress/gutenberg/blob/trunk/.git-blame-ignore-revs
> 
> Let's remember about the follow-up suggested by @noisysocks.

Added with d25c39d46adb1efdfb1a4ba2a834539bdfd6bdc9.

I have just noticed triple `t` in the code comment 😱

**Pull request merged**